.ci/test-local-tap.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+set -eo pipefail
+
+. "${0%/*}/libfunctions.sh"
+
+echo '--- :evergreen_tree: Configuring Environment'
+exec 9>&1 # enable tee to write to STDOUT as a file
+test_bottle
+
+echo '+++ :beer: Homebrew Tap and Install'
+perform 'brew update'
+if [[ ! $(python3 --version 2>/dev/null) ]]; then
+   perform 'brew install python3'
+fi
+perform "sed -i.bk -e 's/url \".*\"/url \"http:\/\/127.0.0.1:7800\"/' ${BUILDKITE_PIPELINE_SLUG}/*.rb"
+perform "pushd ${BUILDKITE_PIPELINE_SLUG} && git add *.rb && git commit -m 'test it!' && popd"
+perform "brew tap eosio/eosio ${BUILDKITE_PIPELINE_SLUG}"
+perform '{ python3 -m http.server 7800 & } && export HTTP_SERVER_PID=$!'
+perform 'sleep 20s'
```

Comment:
What if, instead of a static `sleep` value that could be too short or too long in-context, we have a little loop that waits for the server to come up. Maybe something like this?
```bash
echo 'Waiting for HTTP server to come online...'
while [[ ! $(curl -fsL 'http://localhost:7800' > /dev/null) ]]; do sleep 1; done
echo 'HTTP server is up!'
```
I did a little test in my shell and it seems to be working alright.
<img width="701" alt="Screen Shot 2021-08-04 at 12 25 33 PM" src="https://user-images.githubusercontent.com/34947245/128218147-303c35e0-e195-4ac5-b896-1839d3823ef7.png">

Reply:
We can if-and-only-if the server being "up" means that its ready to "serve" traffic. Polling is always a better solution.